File: src/main/java/dk/fitfit/oneliner/service/OneLinerService.java
package dk.fitfit.oneliner.service;

import dk.fitfit.oneliner.domain.OneLiner;
import dk.fitfit.oneliner.repository.OneLinerRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Set;

@Service
public class OneLinerService implements OneLinerServiceInterface {
	private final OneLinerRepository oneLinerRepository;

	public OneLinerService(OneLinerRepository oneLinerRepository) {
		this.oneLinerRepository = oneLinerRepository;
	}

	@Override
	public Set<OneLiner> findByAllTags(Set<String> tags) {
		return oneLinerRepository.findByAllTags(tags, tags.size());
	}

	@Override
	public List<OneLiner> findAll() {
		return oneLinerRepository.findAll();
	}

	@Override
	public OneLiner save(OneLiner oneLiner) {
		return oneLinerRepository.save(oneLiner);
	}

	@Override
	public OneLiner getOne(long id) {
		return oneLinerRepository.getOne(id);
	}
}

File: src/main/java/dk/fitfit/oneliner/service/OneLinerService.kt
package dk.fitfit.oneliner.service

import dk.fitfit.oneliner.domain.OneLiner
import dk.fitfit.oneliner.repository.OneLinerRepository
import org.springframework.stereotype.Service

@Service
class OneLinerService(private val oneLinerRepository: OneLinerRepository) : OneLinerServiceInterface {

    override fun findByAllTags(tags: Set<String>): Set<OneLiner> {
        return oneLinerRepository.findByAllTags(tags, tags.size.toLong())
    }

    override fun findAll(): List<OneLiner> {
        return oneLinerRepository.findAll()
    }

    override fun save(oneLiner: OneLiner): OneLiner {
        return oneLinerRepository.save(oneLiner)
    }

    override fun getOne(id: Long): OneLiner {
        return oneLinerRepository.getOne(id)
    }
}
